hfile.c

```diff
@@ -1035,15 +1035,20 @@ static int init_add_plugin(void *obj, int (*init)(struct hFILE_plugin *),
     return 0;
 }
 
-static void load_hfile_plugins()
+/*
+ * Returns 0 on success,
+ *        <0 on failure
+ */
+static int load_hfile_plugins()
 {
     static const struct hFILE_scheme_handler
         data = { hopen_mem, hfile_always_local, "built-in", 80 },
         file = { hopen_fd_fileuri, hfile_always_local, "built-in", 80 },
         preload = { hopen_preload, is_preload_url_remote, "built-in", 80 };
 
     schemes = kh_init(scheme_string);
-    if (schemes == NULL) abort();
+    if (schemes == NULL)
```

Comment:
(At the time, I judged there was little point in extreme malloc error handling here.) This change is incomplete without also checking the `kh_put` in `hfile_add_scheme_handler()`.

Reply:
I'm aware there are many more things to fix, such as all the khash and kstring checks.  However my strategy is as follows (ignoring threading for this argument), using a malloc fuzzer that fails on the Nth allocation.

1. Count up from 1 to X to find the value of X where the exit code is not 1.  It should be zero, but some also crashed.
2. Count from X upwards to validate that all subsequent exist codes should be 0.  Again, I found cases where this failed as we erroneously returned success when we'd infact failed.

Now surprisingly for 2, this doesn't happen often, even on VCF which is riddled with kstring calls without checks.  The reason being that for all the missed errors, it eventually does something which does check and so blows up a bit later.  Hence the number of places to fix was small.

This was one example which made my testing logic problematic, so I fixed it so I could continue with the other tests.  Step in the right direction and all that.

FWIW this PR makes things considerably more stable in memory tight situations, but so far only for reading (both single and multi-threaded).  I've just started on the writing side.  Why do I care?  Because we very occasionally get baffling bugs reported from people who claim to have dozens of GB of memory available, only to discover they're using a batch submission system which limited the memory.  Ideally the tools should be reporting out of memory too, and they do now more often than before, but that's still not universally true.  (It also turns out the bug we were hunting this time wasn't due to that, but it's still helpful to discount that.)